microsoftgraph/microsoft-graph-toolkit
# Add presence badge to person avatar

Closes #268 

### PR Type
- Feature

### Description of the changes
- Please find design [here](https://www.figma.com/file/tVj3DubCqeip6mhw14Up0j/Microsoft-Graph-Toolkit-(main)?node-id=723%3A16751).
- Colors added to mgt-person.css comes from [Fluent UI](https://developer.microsoft.com/en-us/fluentui#/styles/web/colors/presence)
- Presence API [reference docs](https://docs.microsoft.com/en-us/graph/api/presence-get?view=graph-rest-beta&tabs=http).

### How to test:
You can run storybook and see the list of status in [Person Presence Display All](http://localhost:6006/?path=/story/components-mgt-person--person-presence-display-all)
<img width="346" alt="Screen Shot 2020-04-21 at 1 56 25 PM" src="https://user-images.githubusercontent.com/7429891/79913101-eae05880-83d7-11ea-8c41-0c19e43a2edf.png">

You can also `npm start` and replace `body` with the following in `index.html` file
```
  <body>
    <mgt-teams-provider
      client-id="a974dfa0-9f57-49b9-95db-90f04ce2111a"
      auth-popup-url="auth.html"
    ></mgt-teams-provider>

    <mgt-msal-provider
      client-id="a974dfa0-9f57-49b9-95db-90f04ce2111a"
      scopes="user.read,people.read,user.readbasic.all,contacts.read,calendars.read,presence.read"
      depends-on="mgt-teams-provider"
    ></mgt-msal-provider>

    

    <mgt-login></mgt-login>
    <mgt-person person-query="me" show-name show-email person-card="hover" show-presence></mgt-person>
    <mgt-agenda></mgt-agenda>
    <mgt-people
      user-ids="91ca32bf-43f4-4907-8ad5-bec920fb4ad3,a2e824c2-acb9-456b-b159-4c25fdd3c472,0cd426de-cb7a-41df-a51b-d6d189fafcd2,a295d82d-c88e-4f11-a5b3-5e1f7b0da018"
      show-presence
    ></mgt-people>
  </body>
```
### PR checklist
- [x] Project builds (`npm run build`) and changes have been tested in supported browsers (including IE11)
- [x] All public classes and methods have been documented
- [x] Added appropriate [documentation](https://github.com/microsoftgraph/microsoft-graph-docs/tree/master/concepts/toolkit) [target branch `mgt/next` for new features]. Docs PR: https://github.com/microsoftgraph/microsoft-graph-docs/pull/7857
- [x] License header has been added to all new source files (`npm run setLicense`)
